packages/clerk-js/src/ui/signUp/SignUpContinue.tsx

```diff
@@ -0,0 +1,161 @@
+  const { displayConfig, userSettings } = environment;
+  const { setSession } = useCoreClerk();
+  const { navigateAfterSignUp } = useSignUpContext();
+  const [activeCommIdentifierType, setActiveCommIdentifierType] = React.useState<ActiveIdentifier>('emailAddress');
+  const signUp = useCoreSignUp();
+
+  // Redirect to sign-up if there is no persisted sign-up
+
+  if (!signUp.id) {
+    void navigate(displayConfig.signUpUrl);
+    return null;
+  }
+
+  // Pre-populate fields from existing sign-up object
+
+  const formState = {
+    firstName: useFieldState('first_name', signUp.firstName || ''),
+    lastName: useFieldState('last_name', signUp.lastName || ''),
+    emailOrPhone: useFieldState('email_or_phone', ''), // this is only to satisfy the type declaration
+    emailAddress: useFieldState('email_address', signUp.emailAddress || ''),
+    username: useFieldState('username', signUp.username || ''),
+    phoneNumber: useFieldState('phone_number', signUp.phoneNumber || ''),
+    password: useFieldState('password', ''),
+    ticket: useFieldState('ticket', ''),
+  } as const;
+
+  type FormStateKey = keyof typeof formState;
+
+  const [error, setError] = React.useState<string | undefined>();
+
+  const hasEmail = !!formState.emailAddress.value;
+  const hasVerifiedExternalAccount = signUp.verifications?.externalAccount?.status == 'verified';
+  const hasVerifiedWeb3 = signUp.verifications?.web3Wallet?.status == 'verified';
+
+  const fields = determineActiveFields({
+    environment,
+    hasEmail,
+    activeCommIdentifierType,
+    signUp,
+  });
+
+  const oauthOptions = userSettings.socialProviderStrategies;
+  const web3Options = userSettings.web3FirstFactors;
+
+  const handleChangeActive = (type: ActiveIdentifier) => (e: React.MouseEvent) => {
+    e.preventDefault();
+
+    if (!fields.emailOrPhone) {
+      return;
+    }
+
+    setActiveCommIdentifierType(type);
+  };
+
+  const handleSubmit = async (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+
+    const fieldsToSubmit = Object.entries(fields).reduce(
+      (acc, [k, v]) => [...acc, ...(v && formState[k as FormStateKey] ? [formState[k as FormStateKey]] : [])],
+      [] as Array<FieldState<any>>,
+    );
+
+    try {
+      setError(undefined);
+      const req = buildRequest(fieldsToSubmit);
+      const res = await signUp.update(req);
+      return completeSignUpFlow(res);
+    } catch (err) {
+      handleError(err, fieldsToSubmit, setError);
+    }
+  };
+
+  const completeSignUpFlow = (su: SignUpResource) => {
+    if (su.status === 'complete') {
+      return setSession(su.createdSessionId, navigateAfterSignUp);
+    } else if (su.emailAddress && su.verifications.emailAddress.status !== 'verified') {
+      return navigate(displayConfig.signUpUrl + '/verify-email-address');
+    } else if (su.phoneNumber && su.verifications.phoneNumber.status !== 'verified') {
+      return navigate(displayConfig.signUpUrl + '/verify-phone-number');
```

Comment:
🙃 
+ gives me these errors:
![image](https://user-images.githubusercontent.com/15251081/167794625-47325675-2e2e-4fb9-8940-4b1ebd765fce.png)

Reply:
And `signUpUrl` is supposed to be a string, go figure.